embeddings/evaluator/leaderboard.py

```diff
@@ -0,0 +1,111 @@
+from enum import Enum
+from typing import Dict, Final, List
+
+
+class LeaderboardDataset(str, Enum):
+    abusive_clauses = "Abusive Clauses"
+    aspectemo = "AspectEmo"
+    cdsc_e = "CDSC-E"
+    dialogue_acts = "Dialogue Acts"
+    dyk = "DYK"
+    kpwr_ner = "KPWr NER"
+    nkjp_pos = "NKJP POS"
+    polemo2 = "PolEmo 2.0"
+    polemo2_in = "PolEmo 2.0 (In-domain)"
+    polemo2_out = "PolEmo 2.0 (Out-domain)"
+    political_advertising = "Political Advertising"
+    psc = "PSC"
+    punctuation_restoration = "Punctuation Restoration"
+
+
+class LeaderboardTask(str, Enum):
+    abusive_clauses_detection = "Abusive Clauses Detection"
+    aspect_based_sentiment_analysis = "Aspect-based Sentiment Analysis"
+    dialogue_acts_classification = "Dialogue Acts Classification"
+    entailment_classification = "Entailment Classification"
+    ner = "Named Entity Recognition"
+    paraphrase_classification = "Paraphrase Classification"
+    political_advertising_detection = "Political Advertising Detection"
+    pos = "Part-of-speech Tagging"
+    punctuation_restoration = "Punctuation Restoration"
+    qa_classification = "Q&A Classification"
+    sentiment_analysis = "Sentiment Analysis"
+
+
+class LeaderboardDomain(str, Enum):
```

Comment:
Can we change it and assign domain such as 

1. News - News articles from a variety of sources and topics, such as politics, sports, business, and entertainment.
2. Social Media - Social media posts from various platforms, such as Twitter, Facebook, and Instagram, to capture informal language and user-generated content.
3. Conversational - Conversational text, such as chat logs or customer service transcripts, to capture natural language interactions.
4. Scientific and technical - Scientific and technical documents from various fields, such as medicine, engineering, and law, to capture domain-specific language and terminology.
5. Literary - Literary texts, such as books, poems, and plays, to capture creative writing styles and nuances.
6. Legal - Legal texts can include various types of documents such as contracts, statutes, regulations, court opinions, and legal briefs. They may also contain specific features such as citations, references to legal cases, and specialized terminology.
7. Other - if anything does not fit.

Wikipedia looks more like source not a genre.

Reply:
It is based on the LEPISZCZE paper taxonomy https://arxiv.org/pdf/2211.13112.pdf (Table 1)

Reply:
I know, and I am not sure it is a good taxonomy. :D

Reply:
I do not have any taxonomy in my mind right now. We can start with the example described above.